From c878b5c74967b62df57b0ffce1d301c186494382 Mon Sep 17 00:00:00 2001
From: Richard Su <rwsu@redhat.com>
Date: Fri, 4 Apr 2014 20:39:57 -0700
Subject: [PATCH] Update swift's selinux policies

Allow swift permission to /mnt/state/var/cache/swift.
Allow rsync to write to /mnt/state/var/log/rsyncd.log.

Move directory creation from post-configure.d to configure.d.

Change-Id: I9c8730be8d880229ec3224690f46406d55b2e8e1
---
 .../os-refresh-config/configure.d/10-swift-storage-state      |  5 +++++
 .../os-refresh-config/configure.d/20-swift-storage-selinux    | 11 +++++++++++
 .../os-refresh-config/post-configure.d/74-swift-storage       |  2 --
 3 files changed, 16 insertions(+), 2 deletions(-)
 create mode 100755 elements/swift-storage/os-refresh-config/configure.d/10-swift-storage-state
 create mode 100755 elements/swift-storage/os-refresh-config/configure.d/20-swift-storage-selinux

diff --git a/elements/swift-storage/os-refresh-config/configure.d/10-swift-storage-state b/elements/swift-storage/os-refresh-config/configure.d/10-swift-storage-state
new file mode 100755
index 0000000..558d758
--- /dev/null
+++ b/elements/swift-storage/os-refresh-config/configure.d/10-swift-storage-state
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -eu
+
+[ -d /mnt/state/var/cache/swift ] && exit 0
+install -d -o swift -g swift /mnt/state/var/cache/swift
diff --git a/elements/swift-storage/os-refresh-config/configure.d/20-swift-storage-selinux b/elements/swift-storage/os-refresh-config/configure.d/20-swift-storage-selinux
new file mode 100755
index 0000000..261b157
--- /dev/null
+++ b/elements/swift-storage/os-refresh-config/configure.d/20-swift-storage-selinux
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -eu
+
+[ -x /usr/sbin/semanage ] || exit 0
+
+semanage fcontext -a -t swift_var_cache_t "/mnt/state/var/cache/swift(/.*)?"
+restorecon -Rv /mnt/state/var/cache/swift
+
+# allows rsync to write to /mnt/state/var/log/rsyncd.log
+semanage fcontext -a -t var_log_t "/mnt/state/var/log"
+restorecon -Rv /mnt/state/var/log
diff --git a/elements/swift-storage/os-refresh-config/post-configure.d/74-swift-storage b/elements/swift-storage/os-refresh-config/post-configure.d/74-swift-storage
index 3f57936..dd178c2 100755
--- a/elements/swift-storage/os-refresh-config/post-configure.d/74-swift-storage
+++ b/elements/swift-storage/os-refresh-config/post-configure.d/74-swift-storage
@@ -1,8 +1,6 @@
 #!/bin/bash
 set -eu
 
-[ -d /mnt/state/var/cache/swift ] || install -d -o swift -g swift /mnt/state/var/cache/swift
-
 os-svc-enable -n swift-account
 os-svc-enable -n swift-account-auditor
 os-svc-enable -n swift-account-reaper
-- 
1.9.0

